pulsar-broker/src/test/java/org/apache/pulsar/broker/service/BacklogQuotaManagerTest.java

```diff
@@ -475,10 +477,12 @@ public void testConsumerBacklogEvictionTimeQuota() throws Exception {
         rolloverStats();
 
         stats = admin.topics().getStats(topic1);
+        PersistentTopic topic1Reference = (PersistentTopic) pulsar.getBrokerService().getTopicReference(topic1).get();
+        ManagedLedgerImpl ml = (ManagedLedgerImpl) topic1Reference.getManagedLedger();
         // Messages on first 2 ledgers should be expired, backlog is number of
-        // message in current ledger which should be 4.
-        assertEquals(stats.getSubscriptions().get(subName1).getMsgBacklog(), 4);
-        assertEquals(stats.getSubscriptions().get(subName2).getMsgBacklog(), 4);
+        // message in current ledger.
+        assertEquals(stats.getSubscriptions().get(subName1).getMsgBacklog(), ml.getNumberOfEntries());
```

Comment:
This is an interesting problem, considering the reporting issue failed with an actual value of 5. Since we produce 14 messages, that means one of the other two ledgers for this test only had 4 entries.

What if we also assert that `ml.getNumberOfEntries() >= 4` and also assert that `ml.getLedgersInfoAsList().size() == 1`? That way we don't lose too much precision in the test?

Reply:
I think the assert is enough for now.

Reply:
@michaeljmarshall

Reply:
How would this verify the `BacklogQuota RetentionPolicy` like this ? 
I think we should use `ml.getCurrentLedgerEntries` here?

Reply:
Yes, getCurrentLedgerEntries is more precise